server/planning/search/eventsplanning_filters.py

```diff
@@ -12,38 +12,200 @@
 
 """
 
+from typing import NamedTuple
+
 from copy import deepcopy
 from eve.utils import config
 import logging
 import superdesk
 from superdesk import Resource
-from superdesk.errors import SuperdeskApiError
 from superdesk.notification import push_notification
+from superdesk.resource import not_analyzed
+from superdesk.metadata.item import metadata_schema
 from apps.auth import get_user_id
-from planning.common import set_original_creator
+from planning.common import set_original_creator, SPIKED_STATE
+from planning.events.events_schema import events_schema
+from planning.planning.planning import planning_schema
 
 
 logger = logging.getLogger(__name__)
 endpoint = 'events_planning_filters'
 
+
+#: item types
+class ItemTypes(NamedTuple):
+    EVENT: str
+    PLANNING: str
+    COMBINED: str
+
+
+ITEM_TYPES: ItemTypes = ItemTypes('events', 'planning', 'combined')
+
+
 filters_schema = {
     'name': {
         'type': 'string',
         'iunique': True,
         'required': True
     },
-    'agendas': {
-        'type': 'list',
-    },
-    'calendars': {
-        'type': 'list',
-    },
-    'places': {
-        'type': 'list',
+    'item_type': {
+        'type': 'string',
+        'allowed': tuple(ITEM_TYPES),
+        'default': ITEM_TYPES.COMBINED,
+        'mapping': not_analyzed,
```

Comment:
not sure why there is a mapping here, does it get saved to elastic?

Reply:
Elastic is not used on this resource. Considering mappings are added here through the use of `metadata_schema` I decided to add mappings everywhere for this schema.

I could remove all manual mappings if you think that's better (and just leave the mappings added through `metadata_schema`)?

Reply:
yep that should be fine